File: remove-element/remove-element.cs
public class Solution {
    public int RemoveElement(int[] nums, int val) {
        var len = nums.Length;

        for (var i = 0; i < len; i++)
        {
            if (nums[i] == val)
            {
                // i =2 len = 7 
                do
                {
                    if (i == len - 1)
                    {
                        return len - 1;
                    }
                    nums[i] = nums[len - 1];
                } while (nums[len-- - 1] == val);

            }
        }
        
        return len;
    }
}

File: remove-element/remove-element.cpp
class Solution {
public:
    int removeElement(vector<int> &nums, int val)
    {
        int result = 0;

        for (int i = 0; i < nums.size(); i++)
        {
            if (nums[i] != val)
            {
                result++;
            }
            else
            {
                for (int j = i + 1; j < nums.size(); j++)
                {
                    if (nums[j] != val)
                    {
                        swap(nums[i], nums[j]);
                        result++;
                        break;
                    }
                }
            }
        }
        return result;
    }
};